src/Microsoft.OpenApi/Services/OpenApiUrlTreeNode.cs

```diff
@@ -0,0 +1,225 @@
+        /// Appends a path and the PathItem to an <see cref="OpenApiUrlTreeNode"/> node.
+        /// </summary>
+        /// <param name="path">An OpenAPI path.</param>
+        /// <param name="pathItem">Path Item object that describes the operations available on an OpenAPI path.</param>
+        /// <param name="label">A name tag for labelling the <see cref="OpenApiUrlTreeNode"/> node.</param>
+        /// <returns>An <see cref="OpenApiUrlTreeNode"/> node describing an OpenAPI path.</returns>
+        public OpenApiUrlTreeNode Attach(string path,
+                                         OpenApiPathItem pathItem,
+                                         string label)
+        {
+            Utils.CheckArgumentNullOrEmpty(label, nameof(label));
+
+            if (path.StartsWith("/"))
+            {
+                // Remove leading slash
+                path = path.Substring(1);
+            }
+
+            var segments = path.Split('/');
+
+            return Attach(segments: segments,
+                          pathItem: pathItem,
+                          label: label,
+                          currentPath: "");
+        }
+
+        /// <summary>
+        /// Assembles the constituent properties of an <see cref="OpenApiUrlTreeNode"/> node.
+        /// </summary>
+        /// <param name="segments">IEnumerable subdirectories of a relative path.</param>
+        /// <param name="pathItem">Path Item object that describes the operations available on an OpenAPI path.</param>
+        /// <param name="label">A name tag for labelling the <see cref="OpenApiUrlTreeNode"/> node.</param>
+        /// <param name="currentPath">The relative path of a node.</param>
+        /// <returns>An <see cref="OpenApiUrlTreeNode"/> node with all constituent properties assembled.</returns>
+        private OpenApiUrlTreeNode Attach(IEnumerable<string> segments,
+                                          OpenApiPathItem pathItem,
+                                          string label,
+                                          string currentPath)
+        {
+            var segment = segments.FirstOrDefault();
+            if (string.IsNullOrEmpty(segment))
+            {
+                if (PathItems.ContainsKey(label))
+                {
+                    throw new ArgumentException("A duplicate label already exists for this node.", nameof(label));
+                }
+
+                Path = currentPath;
+                PathItems.Add(label, pathItem);
+                return this;
+            }
+
+            // If the child segment has already been defined, then insert into it
+            if (Children.ContainsKey(segment))
+            {
+                return Children[segment].Attach(segments: segments.Skip(1),
+                                                pathItem: pathItem,
+                                                label: label,
+                                                currentPath: currentPath + "\\" + segment);
```

Comment:
consider using a constant for the separator value instead of duplicating the definition